### PR TITLE
Task 26 - Add method to returns most popular short url

### PR DESCRIPTION
**Pull request by RobertoJavaDev**

### CODE REVIEW

File: src/main/java/pl/stormit/tinyurl/controller/UrlAnalyticsController.java [hunk 1]
```diff
@@ -32,4 +34,15 @@ public ResponseEntity<List<UrlAnalytics>> getAllAnalytics() {
         List<UrlAnalytics> urlAnalyticsDtoList = urlAnalyticsService.getAllAnalytics();
         return new ResponseEntity<>(urlAnalyticsDtoList, HttpStatus.OK);
     }
+
+    @GetMapping("/most-popular")
+    public ResponseEntity<List<UrlDto>> getMostPopularUrls() {
+
```
Comment (StormITpl) [inline src/main/java/pl/stormit/tinyurl/controller/UrlAnalyticsController.java:40]: redundant new line

File: src/main/java/pl/stormit/tinyurl/service/UrlAnalyticsService.java [hunk 2]
```diff
@@ -71,4 +82,22 @@ private Long checkClicksAmountOnShortUrl(UUID urlId) {
             return incMaxClickValue;
         }
     }
+
+    public List<UrlDto> findMostPopularUrls() {
+        PageRequest pageRequest = PageRequest.of(0, 3);
```
Comment (StormITpl) [inline src/main/java/pl/stormit/tinyurl/service/UrlAnalyticsService.java:87]: Move it to some configuration or at least to a constant.

File: src/main/java/pl/stormit/tinyurl/service/UrlAnalyticsService.java [hunk 3]
```diff
@@ -71,4 +82,22 @@ private Long checkClicksAmountOnShortUrl(UUID urlId) {
...
+        List<MostPopularUrlResult> results = urlRepository.findMostPopularUrls(pageRequest);
+
+        List<UrlDto> popularUrls = new ArrayList<>();
+        for (MostPopularUrlResult result : results) {
```
Comment (StormITpl) [inline src/main/java/pl/stormit/tinyurl/service/UrlAnalyticsService.java:91]: Looks like a lot of logic.
Think how to simplify it, or at least move it to a separate method/class.

File: src/test/java/pl/stormit/tinyurl/controller/UrlControllerTest.java [hunk 4]
```diff
@@ -69,9 +69,9 @@ void shouldReturnStatusCreatedWhenCreateShortUrlCorrectly() throws Exception {
     @Test
     void shouldThrowAnExceptionWhenLongUrlIsNull() throws Exception {
         //given
-        urlDto = new UrlDto(null, "");
+        urlDto = new UrlDto(null, "", null);
```
Comment (StormITpl) [inline src/test/java/pl/stormit/tinyurl/controller/UrlControllerTest.java:72]: Maybe it's worth to add a new constuctor with some default values.

**Review by StormITpl — APPROVED** [review]

File: src/main/java/pl/stormit/tinyurl/controller/UrlAnalyticsController.java (hunk 1 above)
Comment (RobertoJavaDev) [reply src/main/java/pl/stormit/tinyurl/controller/UrlAnalyticsController.java:40]: Corrected

File: src/main/java/pl/stormit/tinyurl/service/UrlAnalyticsService.java (hunk 2 above)
Comment (RobertoJavaDev) [reply src/main/java/pl/stormit/tinyurl/service/UrlAnalyticsService.java:87]: Moved to constant

File: src/test/java/pl/stormit/tinyurl/controller/UrlControllerTest.java (hunk 4 above)
Comment (RobertoJavaDev) [reply src/test/java/pl/stormit/tinyurl/controller/UrlControllerTest.java:72]: I will correct this in next task - refactoring

**Pull request merged**